testing_utilities/check_well_formedness.hpp

```diff
@@ -0,0 +1,30 @@
+#pragma once
+#include <type_traits>
+
+#define PRINCIPIA_CONCATENATE(s1, s2) s1##s2
+#define PRINCIPIA_CONCATENATE_EXPANSION(s1, s2) PRINCIPIA_CONCATENATE(s1, s2)
+
+#define PRINCIPIA_CHECK_WELL_FORMEDNESS(unique_concept_name,                  \
+                                        expected_well_formedness,             \
+                                        expected_well_formedness_description, \
+                                        expression,                           \
+                                        ...)                                  \
+  template<template<typename> typename WITH = std::type_identity_t>           \
```

Comment:
The uppercase `WITH` is rather tacky.  We have the precedent of `with_evaluator`, let's call this one `with_parameter`.

Reply:
`with_variable`, as discussed.